src/libPMacc/include/particles/memory/dataTypes/Particle.hpp

```diff
@@ -139,20 +141,21 @@ struct Particle : public InheritLinearly<typename T_FrameType::MethodsList>
 namespace traits
 {
 
-template<typename T_Key,
-typename T_FrameType
->
-struct HasIdentifier<
-PMacc::Particle<T_FrameType>,
-T_Key
->
-{
-private:
-    typedef T_FrameType FrameType;
-public:
-    typedef typename HasIdentifier<FrameType, T_Key>::type type;
```

Comment:
It look like the old implementation of HasIdentifier for a particle is wrong implemented. If HasIdentifier<deselect<Particle,X>,X> is called it returns true but should return false if the identifier is already deselected.
I will check this tomorrow again.